src/lib/nodes/abstract.client.ts

```diff
@@ -95,7 +95,11 @@ export abstract class Client<N extends Node> {
   // This method can throw an error if there are no online nodes.
   // Better use "useClient()" method.
   getClient(): N['client'] {
-    return this.getNode().client
+    const node = this.getNode()
+
+    // Ethereum nodes return a new client every time depending on `preferDomain` property.
+    if (node.type === 'eth') return (node.client as () => N['client'])()
+    else return node.client
   }
```

Comment:
Looks like workaround. Just deprecate the method if it's not used anywhere.

Reply:
Well, at least it is used [here](https://github.com/Adamant-im/adamant-im/blob/dev/src/store/modules/erc20/erc20-actions.js#L55) and [here](https://github.com/Adamant-im/adamant-im/blob/dev/src/store/modules/erc20/erc20-actions.js#L106):
![Screenshot from 2025-06-21 08-16-39](https://github.com/user-attachments/assets/4fd2e2fa-cf7c-47b3-9a26-21ec6b0ee197)
I would not deprecate it in this PR.
Let's move it to `src/lib/nodes/eth/EthClient.ts` first